### PR TITLE
DB-6635: Support ldap group as authenticated user (2.5)

### PR DESCRIPTION
**Pull request by snowspeeder**
Key changes: fetch the authenticated group user name from ldap, and save it as a groupser. If the groupuser is same as as the database owner, permissions checks are passed.
Tested manually with the ldap server that with group user 'cn=splice' in the ldap with user, details in jira. (also see EE change)

### CODE REVIEW

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementTablePermission.java [hunk 1]
```diff
@@ -220,6 +220,10 @@ protected boolean hasPermissionOnTable(LanguageConnectionContext lcc,
 							Authorizer.PUBLIC_AUTHORIZATION_ID,
 							dbo);
 				}
+				if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementTablePermission.java]: The current sequence of permission checks in the function hasPermissionOnTable() is the following:
1. whether PUBLIC has permission on this database object
2. if NONE/UNAUTHORIZED, whether the currentUser has permission on this database object
3 if NONE, whether PUBLIC has schema level permission for the schema this database object belongs to.
4. if NONE/UNAUTHORIZED, whether currentUser has schema level permission for the schema this database object belongs to.
5. If NONE, check the role permissions from PUBLIC/currentUser.

With the addition of CurrentGroupUser, In addition to the check for roles from the currentGroupUser, I think we also need to add check
2.5, If NONE, whether the currentGroupUser has permission on this database object
4.5 If NONE, whether currentGroupUser has schema level permission for the schema this database object belongs to.

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java [hunk 2]
```diff
@@ -186,6 +186,10 @@ protected boolean hasPermissionOnSchema(LanguageConnectionContext lcc,
 							Authorizer.PUBLIC_AUTHORIZATION_ID,
 							dbo);
 				}
+				if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java]: Similar to StatementTablePermission.java, we need to add the following before we check role permissions:
If NONE, whether currentGroupUser has schema level permission for the schema this database object belongs to.

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementPermission.java [hunk 3]
```diff
@@ -159,6 +159,10 @@ public String getObjectType()
 					Authorizer.PUBLIC_AUTHORIZATION_ID,
 					dbo);
 			}
+			if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementPermission.java]: Similar to StatementSchemaPermission.java, we need to add check 
		if( !isCorrectPermission( perm ) ) { perm = getPermissionDescriptor(lcc.getCurrentGroupUser(activation), dd ); }

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementColumnPermission.java [hunk 4]
```diff
@@ -115,7 +115,7 @@ public void check( LanguageConnectionContext lcc,
 
         String currentUserId = lcc.getCurrentUserId(activation);
 
-		FormatableBitSet permittedColumns = null;
+			FormatableBitSet permittedColumns = null;
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementColumnPermission.java]: When populating the permittedColumns, in addition to the call of addPermittedColumns() for public and currentUserId, I think we should add the call for currentGroupUser too.

**Comment by snowspeeder** [comment]
Thanks Yi! I've added those checks for the groupuser.

**Pull request merged**